### PR TITLE
Fix Gaussian splats not rendering in Standalone game

### PR DESCRIPTION
**Pull request by UmarMukhtar0039**
## Description

Gaussian splats never renders in the Standalone game session. There are two main causes:

- `GetPrimaryWorld()` lacked proper handling for picking the right world context for all play modes (PIE, Standalone etc.) it only checked `GEditor`, which is null at runtime in a game session. This fix ensures that the code gracefully falls through to `GEngine` and prefers the Game mode world context.

- The splat actor was spawned with `bTemporaryEditorActor`, making it an editor-preview actor. This supresses `BeginPlay` in a game world, so the attached Niagara system never initialized its per instance data. It's now only marked as a preview actor in a non-game (editor) worlds.

## Issue number or link

N/A

## Author checklist

- [x] I have submitted a [Contributor License Agreement](https://github.com/CesiumGS/community/tree/main/CLAs) (only needed once).
- [x] I have done a full self-review of my code.
- [x] I have updated `CHANGES.md` with a short summary of my change (for user-facing changes).
- ~~[ ] I have added or updated unit tests to ensure consistent code coverage as necessary.~~
- ~~[ ] I have updated the documentation as necessary.~~

## Testing plan

1. Launch an Unreal project with Gaussian splats enabled (preferrably cesium-unreal-samples).
2. Open the map with Gaussian splats
3. Launch in Standalone mode - splats now should render. 

## Reviewer checklist

Thank you for taking the time to review this PR. By approving a PR you are taking as much responsibility for these changes as the author.

As you review, please go through the checklist below:

- [ ] Review and run all parts of the test plan on this branch and verify it matches expectations.
    - [ ] If the issue is a bug please make sure you can reproduce the bug in the main branch and then checkout this branch to make sure it actually solved the issue.
- [ ] Review the code and make sure you do not have any remaining questions or concerns. You should understand the code change and the chosen approach. If you are not confident or have doubts about the code, please do not hesitate to ask questions.
    - [ ] Code should follow the [C++ Style Guide](https://cesium.com/learn/cesium-native/ref-doc/style-guide.html)
- [ ] Review the unit tests and make sure there are no missing tests or edge cases.
- [ ] Review documentation changes and updates to `CHANGES.md` to make sure they accurately cover the work in this PR.
- [ ] Verify that the [Contributor License Agreement](https://github.com/CesiumGS/community/tree/main/CLAs) has been submitted, if needed.

### CODE REVIEW

File: Source/CesiumRuntime/Private/CesiumGaussianSplatSubsystem.cpp [hunk 1]
```diff
@@ -65,31 +65,40 @@ FBox CalculateBounds(
...
-  }
-
-  const TIndirectArray<FWorldContext>& Contexts = GEditor->GetWorldContexts();
-#else
-  if (!IsValid(GEngine)) {
-    return nullptr;
-  }
-
-  const TIndirectArray<FWorldContext>& Contexts = GEngine->GetWorldContexts();
+  if (IsValid(GEditor)) {
```
Comment (azrogers) [inline Source/CesiumRuntime/Private/CesiumGaussianSplatSubsystem.cpp:77]: It seems like the change is that, in the editor, if `GEditor` is invalid `GEngine` will be used instead if possible, whereas before in editor `GEngine` was never checked. Is there ever a time where this will be called when `WITH_EDITOR` is defined, `GEngine` is valid, but `GEditor` is not?

Anyways, I'm fine with leaving in the fallback to `GEngine` just in case, but I'm a little wary of the dangling `else` here. Instead I think it would be better to remove the `else` and add a `Contexts == nullptr &&` clause to the `if (IsValid(GEngine))` statement.

**Comment by azrogers** [comment]
@UmarMukhtar0039 Thanks for the fix! Could you fix the issues on CI? At the moment it's just formatting, an `npm install` and `npm run format` needs to be done, but if other jobs come back red those should be fixed too. Besides that it's just the one comment and suggested edit above.

File: Source/CesiumRuntime/Private/CesiumGaussianSplatSubsystem.cpp (hunk 1 above)
Comment (UmarMukhtar0039) [reply Source/CesiumRuntime/Private/CesiumGaussianSplatSubsystem.cpp:77]: Yes, that's exactly the case this fixes. "Is there ever a time where this will be called" - A standalone game session launched from Editor runs as a -game child process, where `GEditor` is null at runtime but `GEngine` is valid. And without this `GetPrimaryWorld()` returns nullptr and splats system never recieves the world.

And you are right about the  dangling `else` sorry for an oversight. I'll make the fix in a little while.

**Comment by UmarMukhtar0039** [comment]
@azrogers this is ready

**Comment by azrogers** [comment]
@UmarMukhtar0039 Thank you! I'll just wait for CI to complete again and if it all looks good I'll get it merged in.

**Comment by UmarMukhtar0039** [comment]
> @UmarMukhtar0039 Thank you! I'll just wait for CI to complete again and if it all looks good I'll get it merged in.

Happy to help
